File: ds/graph/Graph.java
import java.util.ArrayList;
import java.util.List;

public class Graph {
    public static void main(String[] args) {
        List<List<Integer>> graph = new ArrayList<>();
        
        for (int i = 0; i < 5; i++) {
            graph.add(new ArrayList<>());
        }
        
        graph.get(0).add(1);
        graph.get(0).add(4);
        graph.get(1).add(0);
        graph.get(1).add(2);
        graph.get(2).add(1);
        graph.get(2).add(3);
        graph.get(3).add(2);
        graph.get(3).add(4);
        graph.get(4).add(0);
        graph.get(4).add(3);
    }
}

File: ds/graph/graph.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    vector<vector<int>> graph(5);
    
    graph[0].push_back(1);
    graph[0].push_back(4);
    graph[1].push_back(0);
    graph[1].push_back(2);
    graph[2].push_back(1);
    graph[2].push_back(3);
    graph[3].push_back(2);
    graph[3].push_back(4);
    graph[4].push_back(0);
    graph[4].push_back(3);
    
    return 0;
}
